File: challenges/codeeval/easy/232/juanescobar.cs
using System.IO;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        using (StreamReader reader = File.OpenText(args[0]))
        while (!reader.EndOfStream)
        {
            string line = reader.ReadLine();
            if (null == line)
                continue;
            string[] data = line.Split('|');
            string[] numbers = data[0].Trim().Split(' ');
            int iter = System.Int32.Parse(data[1].Trim());
            string temp = "0";
            for (int i=0; i < iter; i++) 
            {
                if (System.Int32.Parse(numbers[i]) > System.Int32.Parse(numbers[i+1])) {
                    temp = numbers[i+1];
                    numbers[i+1] = numbers[i];
                    numbers[i] = temp;
                }
            }

            string output = "";
            for (int i=0; i<numbers.Length; i++) 
            {
                output += " " + numbers[i];
            }
            System.Console.WriteLine(output.Substring(1));
        }
    }
}

File: challenges/codeeval/medium/009/juanescobar.java
import java.io.*;
public class Stack {
    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            String[] list = line.split(" ");
            for (int i = list.length-1; i >= 0; i--) {
                if (i == list.length-1 || i % 2 != 0) {
                    System.out.print(list[i]+" ");
                }
            }
            System.out.println("");
        }
    }
}
